lib/blockchain.go

```diff
@@ -3054,6 +3055,59 @@ func (bc *Blockchain) CreateLikeTxn(
 	return txn, totalInput, changeAmount, fees, nil
 }
 
+func (bc *Blockchain) CreateReactTxn(
+	userPublicKey []byte, postHash BlockHash, isRemove bool, emojiReaction rune,
+	minFeeRateNanosPerKB uint64, mempool *DeSoMempool, additionalOutputs []*DeSoOutput) (
+	_txn *MsgDeSoTxn, _totalInput uint64, _changeAmount uint64, _fees uint64,
+	_err error) {
+
+	//TODO Where would be the best place to place the validation function?
+	// At the moment, only support happy, sad, angry and surprised
+	AcceptedReactions := [4]rune{'😊', '😥', '😠', '😮'}
+	// Validate emoji reaction
+	var isValidEmoji = func(emoji rune) bool {
+		for _, acceptedReaction := range AcceptedReactions {
+			if emoji == acceptedReaction {
+				return true
+			}
+		}
+		return false
+	}
+
+	// TODO Fix bug returning invalid for valid inputs
+	normalizedReaction := norm.NFC.String(string(emojiReaction))
```

Comment:
we need to ensure that the rune is normalized in the connect logic. Anybody could construct the transaction on their own without using our API, so we need to make sure every react emoji is consistent when we're connecting.